query-dsl/src/main/java/org/infinispan/query/dsl/IndexedQueryMode.java

```diff
@@ -0,0 +1,20 @@
+package org.infinispan.query.dsl;
+
+/**
+ * Defines the execution mode of an indexed query.
+ *
+ * @since 9.2
+ */
+public enum IndexedQueryMode {
```

Comment:
Shouldn't we also have an AUTO mode which chooses the best strategy depending on the underlying indexing mode ?

Reply:
👍  I was planning to add it later (another PR)

Reply:
+1 for introducing an AUTO mode. I think that mode should become our default. Do we actually need a value called DEFAULT?

I think the one we now call DEFAULT should be named differently. We need to find a better name that describes in a word that: 'it is executed in a single phase in the caller'.  What is the opposite of clustered/distributed? Can't find a good name right now. :)

And then we could also add a DEFAULT enum value, for convenience, that is just an alias to one of the other constants (and warn users that the DEFAULT might be different in next major ispn version, as we might invent new query mechanisms). :)

Reply:
AUTO will come in a later PR, I possible can squeeze it in before the year ends.

Suggestions for renaming DEFAULT ? 

* SINGLE_PHASE
* SINGLE_STEP
* LOCAL
* NON_BROADCAST :)
* CALLER

Reply:
I've been thinking and all the names above have issues, apart for NON_BROADCAST, but this is silly. I'll go with FETCH which I believe reflects accurately  the fact that the query is tun locally and will read the index, potentially from remote nodes.